src/app/editor/components/create-node-button/create-node-button.component.ts

```diff
@@ -29,8 +36,21 @@ export class CreateNodeButtonComponent {
             );
     }
 
-    itemClick(): void {
-        console.warn('Not implemented');
+    itemClick(schema: SchemaDisplayProperties): void {
+        let askUserToSave: Promise<boolean>;
```

Comment:
This logic might be redundant due to the canDeactivate guard on the detail outlet. We should check this together.